### PR TITLE
renamed `non-indexed labels` stage to `structured metadata`

### PR DESCRIPTION
**Pull request by vlad-diachenko**
#### PR Description

updated loki dependency version to k166 and renamed `non-indexed labels` stage to `structured metadata`

#### Notes to the Reviewer
This PR repeats the changes from https://github.com/grafana/loki/pull/10432

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated
- [x] Config converters updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ Main (unreleased)
 ### Breaking changes
 
 - Set `retry_on_http_429` to `true` by default in the `queue_config` block in static mode's `remote_write`. (@wildum)
+- Renamed `non-indexed-labels` Loki processing stage to `structured_metadata`. (@vlad-diachenko)
```
Comment (rfratto) [inline CHANGELOG.md:16]: Breaking changes should also have an entry in relevant release notes:

- Static mode: https://github.com/grafana/agent/blob/main/docs/sources/static/release-notes.md
- Flow mode: https://github.com/grafana/agent/blob/main/docs/sources/flow/release-notes.md

Can you follow the existing pattern and add an entry matching this change under a new v0.37 header?

Comment (vlad-diachenko) [reply CHANGELOG.md:16]: done ;) thanks

**Review by rfratto — APPROVED** [review]
Thanks, will merge after @clayton-cornell reviews the docs changes.

File: docs/sources/static/release-notes.md [hunk 2]
```diff
@@ -42,6 +42,33 @@ This changed default setting allows the agent to retry sending data when it rece
 * If you explicitly set the `retry_on_http_429`, no action is required.
 * If you do not explicitly set `retry_on_http_429` and you do *not* want to retry on HTTP 429, make sure you set it to `false` when you upgrade to this new version.
 
+### Breaking change: Renamed `non-indexed-labels` Loki processing stage to `structured_metadata`.
+
+It's necessary to renamed Loki processing stage definition `non_indexed_labels` to `structured_metadata` if this stage is used in Agent's config.
```
Comment (clayton-cornell) [inline docs/sources/static/release-notes.md:47]:
```suggestion
If you use the Loki processing stage in your Agent configuration, you must rename the `non_indexed_labels` pipeline stage definition to `structured_metadata`.
```

We need to rephrase this sentence. Is this suggested text still technically accurate?

File: docs/sources/flow/reference/components/loki.process.md [hunk 3]
```diff
@@ -45,29 +45,30 @@ loki.process "LABEL" {
...
+| stage.docker              | [stage.docker][]              | Configures a pre-defined Docker log format pipeline. | no       |
+| stage.drop                | [stage.drop][]                | Configures a `drop` processing stage.                | no       |
+| stage.json                | [stage.json][]                | Configures a JSON processing stage.                  | no       |
+| stage.label_drop          | [stage.label_drop][]          | Configures a `label_drop` processing stage.          | no       |
+| stage.label_keep          | [stage.label_keep][]          | Configures a `label_keep` processing stage.          | no       |
+| stage.labels              | [stage.labels][]              | Configures a labels processing stage.                | no       |
+| stage.structured_metadata | [stage.structured_metadata][] | Configures a structured metadata processing stage.   | no       |
+| stage.limit               | [stage.limit][]               | Configures a `limit` processing stage.               | no       |
+| stage.logfmt              | [stage.logfmt][]              | Configures a logfmt processing stage.                | no       |
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.process.md:59]:
```suggestion
| stage.logfmt              | [stage.logfmt][]              | Configures a `logfmt` processing stage.                | no       |
```

Is logfmt missing the inline code markdown formatting?

File: docs/sources/flow/reference/components/loki.process.md [hunk 4]
```diff
@@ -45,29 +45,30 @@ loki.process "LABEL" {
...
+| Hierarchy                 | Block                         | Description                                          | Required |
+|---------------------------|-------------------------------|------------------------------------------------------| -------- |
+| stage.cri                 | [stage.cri][]                 | Configures a pre-defined CRI-format pipeline.        | no       |
+| stage.docker              | [stage.docker][]              | Configures a pre-defined Docker log format pipeline. | no       |
+| stage.drop                | [stage.drop][]                | Configures a `drop` processing stage.                | no       |
+| stage.json                | [stage.json][]                | Configures a JSON processing stage.                  | no       |
+| stage.label_drop          | [stage.label_drop][]          | Configures a `label_drop` processing stage.          | no       |
+| stage.label_keep          | [stage.label_keep][]          | Configures a `label_keep` processing stage.          | no       |
+| stage.labels              | [stage.labels][]              | Configures a labels processing stage.                | no       |
```
Comment (clayton-cornell) [inline docs/sources/flow/reference/components/loki.process.md:56]:
```suggestion
| stage.labels              | [stage.labels][]              | Configures a `labels` processing stage.                | no       |
```

Is labels missing the inline code markdown formatting?

File: docs/sources/flow/release-notes.md [hunk 5]
```diff
@@ -25,6 +25,28 @@ Other release notes for the different Grafana Agent variants are contained on se
 [release-notes-operator]: {{< relref "../operator/release-notes.md" >}}
 {{% /admonition %}}
 
+
+## v0.37
+
+### Breaking change: Renamed `non-indexed-labels` Loki processing stage to `structured_metadata`.
+
+It's necessary to renamed Loki processing stage definition `non_indexed_labels` to `structured_metadata` if this stage is used in Agent's config.
```
Comment (clayton-cornell) [inline docs/sources/flow/release-notes.md:33]:
```suggestion
If you use the Loki processing stage in your Agent configuration, you must rename the `non_indexed_labels` pipeline stage definition to `structured_metadata`.
```

We need to rephrase this sentence. Is this suggested text still technically accurate?

**Review by clayton-cornell — COMMENTED** [review]
Some suggestions (with comments)

**Comment by tpaschalis** [comment]
cc @vlad-diachenko when you have the time, could you check the doc suggestions and rebase with main?

File: docs/sources/static/release-notes.md [hunk 6]
```diff
@@ -42,6 +42,33 @@ This changed default setting allows the agent to retry sending data when it rece
 * If you explicitly set the `retry_on_http_429`, no action is required.
 * If you do not explicitly set `retry_on_http_429` and you do *not* want to retry on HTTP 429, make sure you set it to `false` when you upgrade to this new version.
 
+### Breaking change: Renamed `non-indexed-labels` Loki processing stage to `structured_metadata`.
```
Comment (thampiotr) [inline docs/sources/static/release-notes.md:45]:
```suggestion
### Breaking change: Renamed `non_indexed_labels` Loki processing stage to `structured_metadata`.
```

File: docs/sources/flow/release-notes.md [hunk 7]
```diff
@@ -25,6 +25,28 @@ Other release notes for the different Grafana Agent variants are contained on se
 [release-notes-operator]: {{< relref "../operator/release-notes.md" >}}
 {{% /admonition %}}
 
+
+## v0.37
+
+### Breaking change: Renamed `non-indexed-labels` Loki processing stage to `structured_metadata`.
```
Comment (thampiotr) [inline docs/sources/flow/release-notes.md:31]:
```suggestion
### Breaking change: Renamed `non_indexed_labels` Loki processing stage to `structured_metadata`.
```

File: CHANGELOG.md (hunk 1 above)
Comment (thampiotr) [inline CHANGELOG.md:16]:
```suggestion
- Renamed `non_indexed_labels` Loki processing stage to `structured_metadata`. (@vlad-diachenko)
```

**Pull request merged**